### DISCUSSION TITLE
Setting up with Docker

### DISCUSSION THREAD

**Discussion by 3raxton** [Freika/dawarich#50] [Q&A] [answered]
Hey @Freika, thanks for the hard work on this! I am interested in using dawarich but I have no background with Docker images and setting up the related elements for successful hosting. 

Is there any insight you can share here on the approach to take or where I may be able to find a guide on setting up a Docker image / container to run dawarich? Thanks!

**Comment by Freika** [comment, marked as answer]
Hi @3raxton! The instructions will vary depending on what platform you want to use to run Dawarich (or any other docker-based app), so it would be easier for you to google something specific for your platform, i.e. "how to start docker container with docker compose on Ubuntu Server".

In general steps would be:

1. Install Docker
2. Start Docker
3. Copy [docker-compose.yml](https://github.com/Freika/dawarich/blob/master/docker-compose.yml) to directory of your choice
4. Run `docker compose up` in Terminal from directory where you placed your docker-compose.yml file

That should basically do the trick :)

**Comment by shanescarlett** [comment]
Have been trying to get it running on a remote headless server (AWS) without much luck. Compose up works well but I can't access port 3000 remotely (refused to connect). Are there any considerations?
docker ps shows that it's running, I believe:
```
CONTAINER ID   IMAGE                     COMMAND                  CREATED        STATUS              PORTS      NAMES
486760380b2e   freikin/dawarich:latest   "dev-entrypoint.sh s…"   46 hours ago   Up About a minute   3000/tcp   dawarich_sidekiq
d6ef2b346a12   redis:7.0-alpine          "docker-entrypoint.s…"   46 hours ago   Up 40 hours         6379/tcp   dawarich-dawarich_redis-1
3d9aee133ea4   postgres:14.2-alpine      "docker-entrypoint.s…"   46 hours ago   Up 40 hours         5432/tcp   dawarich_db
```

**Reply by shanescarlett** [reply]
Sidekiq was trying to connect to rubygems.org through HTTPS so I had to open that up in the EC2 instance's outbound rules, so it looks to be starting up properly now. But :3000 is still inaccessible remotely.

**Reply by Freika** [reply]
Did you experiment with "networks" section in docker-compose.yml? Never tried to run it AWS (or anywhere else, for that matter, except for my home server), so I don't really know what might be the issue, can only assume it's something with network settings

**Reply by shanescarlett** [reply]
Tinkered a bit more, the issue was that I was running this in a nano instance with 512 MiB of memory. Dawarich_app was getting SIGTERM'd from what I assume was a system OOM situation. Spun up a 1GiB instance and it seems to be working fine... Doh!
Maybe hinting at minimum system requirements on the manpage could be helpful? I'm sure a fair few people would like to run this on a minimum system as possible!

**Reply by Freika** [reply]
I'll update Readme, with Hardware requirements section, thank you

**Comment by jonhnet** [comment]
I'm having the same symptoms ("want to try dawarich but cannot start it"), but evidently a different problem: I'm following the extremely simple [two-step README](https://github.com/Freika/dawarich/blob/master/docs/How_to_install_Dawarich_using_Docker.md) to start dawarich for the first time on my home server, and it never converges: it never gets port 3000 open.

This is on `Ubuntu 22.04.4 LTS` with docker `27.2.0`.
The `docker compose logs` are cycling every few minutes, with a 132-line trace that begins
```
dawarich_sidekiq  | Environment: development
dawarich_sidekiq  | dawarich_db (172.19.0.2:5432) open
dawarich_sidekiq  | Latest version already installed. Done.
```
and ends
```
dawarich_sidekiq  | /usr/local/bundle/bin/bundle:25:in `<main>'
dawarich_sidekiq  | Tasks: TOP => data:migrate
dawarich_sidekiq  | (See full trace by running task with --trace)
```

I'm embarrassed to ask because how could I possibly get two steps wrong? But here I am.

This is my docker-compose, which is directly copied without changes from a fresh git clone of the [repo](https://github.com/Freika/dawarich): [docker-compose.yml.txt](https://github.com/user-attachments/files/16826842/docker-compose.yml.txt)

Attached is a log capture in case it's valuable: [logs.txt](https://github.com/user-attachments/files/16826841/logs.txt)

**Reply by Freika** [reply]
@jonhnet based on logs you have attached, I've found a possible bug, which prevents the app from starting. I'll fix it with the next release and let you know.

**Reply by jonhnet** [reply]
Cool, thanks!

I'm very curious what this bug could be, and why it doesn't affect everyone, because my mental model is that starting docker-compose is starting from an empty state; how could my system possibly be in a different state than others'?. I look forward to reading the commit.

**Reply by Freika** [reply]
@jonhnet your problem should be fixed in [0.13.4](https://github.com/Freika/dawarich/releases/tag/0.13.4)

**Reply by jonhnet** [reply]
And it is! Thank you. I see it running now! (My latin game isn't good, but I know a Lorem Ipsum when I see one. :vD)
